src/modules/ProductLicense.rb

```diff
@@ -843,7 +843,8 @@ def GetSourceLicenseDirectory(src_id, fallback_dir)
     # @param [Any] unique ID
     # @param [String] path to directory with unpacked licenses (mandatory)
     def cache_license_acceptance_needed(id, license_dir)
-      raise "Parameter 'license_dir' must not be nil" if license_dir.nil?
+      # license_dir can be nil if there is no license present (e.g. DUDs)
+      return if license_dir.nil?
```

Comment:
I miss a test for this. Is it possible to add an unit test? (If it is not too difficult, I have no idea about testability of this code...)